src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -704,7 +705,7 @@ public object DataContext
 					{protectedModifier} {virtualModifier} void OnDataContextChanged(DependencyPropertyChangedEventArgs e)
 					{{
 						OnDataContextChangedPartial(e);
-						DataContextChanged?.Invoke(this, new DataContextChangedEventArgs(DataContext));
+						DataContextChanged?.Invoke({(typeSymbol.Is(_frameworkElementSymbol) ? "this" : "null")}, new DataContextChangedEventArgs(DataContext));
```

Comment:
DataContext is generated even for non-`FrameworkElement`s. This is still diverging from UWP. But I think it needs more work since FlyoutBase (doesn't derive from `FrameworkElement`) is relying on it.

Reply:
Yes it's still diverging, but we can't work without it at this point.

Reply:
@jeromelaban At least we moved a step towards UWP :)